### PR TITLE
Added Pet related entities

### PR DESCRIPTION
**Pull request by kaboo21**
Approximate time spent: 2 hours

### CODE REVIEW

File: VetClinic.DAL/Configurations/PetEntityConfiguration.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+namespace VetClinic.DAL.Configurations
+{
+    class PetEntityConfiguration : IEntityTypeConfiguration<Pet>
+    {
+        public void Configure(EntityTypeBuilder<Pet> builder)
+        {
+            builder
+                .HasKey(x => x.Id);
+
+
```
Comment (Serafinium) [inline VetClinic.DAL/Configurations/PetEntityConfiguration.cs:15]: please drop redundant space

File: VetClinic.Core/Entities/PetImages.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VetClinic.Core.Entities
+{
+    public class PetImages
```
Comment (Serafinium) [inline VetClinic.Core/Entities/PetImages.cs:7]: Please add Fluent API Configurations

File: VetClinic.BLL/Services/PetServise.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using VetClinic.Core.Interfaces.Services;
+
+namespace VetClinic.BLL.Services
+{
+    public class PetServise:IPetService
```
Comment (Serge1212) [inline VetClinic.BLL/Services/PetServise.cs:8]: add spaces (PetServise : IPetService)

File: VetClinic.Host/Startup.cs [hunk 4]
```diff
@@ -26,14 +26,18 @@ public Startup(IConfiguration configuration)
...
             services.AddScoped(typeof(IRepository<>), typeof(Repository<>));
             services.AddScoped<IEmployeeRepository, EmployeeRepository>();
+            services.AddScoped<IPetRepository, PetRepository>();
+
+
 
             //Services
             services.AddScoped(typeof(IBaseService<>), typeof(BaseService<>));
             services.AddScoped<IEmployeeService, EmployeeService>();
+            services.AddScoped<IPetService, PetServise>();
```
Comment (Serge1212) [inline VetClinic.Host/Startup.cs:40]: Pay attention for namings: ~~PetServise~~ -> PetService

File: VetClinic.WebApi/Controllers/PetsController.cs [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+            if (pet != null)
+            {
+                db.Pets.Remove(pet);
+                db.SaveChanges();
+            }
+            return Ok(pet);
+        }
+
+    }
+}
```
Comment (Serge1212) [inline VetClinic.WebApi/Controllers/PetsController.cs:75]: Do not use DbContext in controllers, user services instead

File: VetClinic.DAL/Context/VetClinicDbContext.cs [hunk 6]
```diff
@@ -1,14 +1,18 @@
...
 using System.Reflection;
+using VetClinic.Core.Entities;
 
 namespace VetClinic.DAL.Context
 {
     public class VetClinicDbContext : DbContext
     { 
         public VetClinicDbContext(DbContextOptions<VetClinicDbContext> options) : base(options)
         {
+            Database.EnsureCreated();
```
Comment (Serge1212) [inline VetClinic.DAL/Context/VetClinicDbContext.cs:11]: Remove this

File: VetClinic.Core/Interfaces/Services/IPetService.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VetClinic.Core.Interfaces.Services
+{
+    public interface IPetService
+    {
+
+    }
```
Comment (Serge1212) [inline VetClinic.Core/Interfaces/Services/IPetService.cs:10]: IPetService must be inherited from IBaseService

**Review by Serge1212 — CHANGES_REQUESTED** [review]

**Pull request merged**